### PR TITLE
[RPA-978] Refactor caching mechanism on the page, blog and single post page

### PR DESCRIPTION
**Pull request by marcin-depcik**
## Description

DONE:
1. Remove code responsible for updating posts each day.
2. Refactor caching mechanism no the home, blog and single post page.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-orchestrator-ui/src/pages/_app.tsx [hunk 1]
```diff
@@ -40,6 +41,12 @@ moment.locale(DEFAULT_LANG);
 // Client-side cache, shared for the whole session of the user in the browser.
 const clientSideEmotionCache = createEmotionCache();
 
+(async function initializeBlogCache() {
+    if (!isCached('en') || !isCached('pl')) {
+        await recreateCache();
+    }
+})();
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/pages/_app.tsx:48]: are you sure it runs at app start?

Comment (preludium) [reply runbotics/runbotics-orchestrator-ui/src/pages/_app.tsx:48]: to me its more like if someone will visit any page of the app

File: runbotics/runbotics-orchestrator-ui/src/pages/blog/index.tsx [hunk 2]
```diff
@@ -45,22 +42,16 @@ const BlogPage: VFC<Props> = ({ posts, categories, tags, page, featuredPost }) =
...
 
-    let cache = getBlogMainCache(locale as Language);
-
-    if (!cache) {
-        cache = await recreateCache(locale as Language);
-
-    } else {
-        res.setHeader('X-Cache', 'HIT');
-    }
-
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/pages/blog/index.tsx:63]: why remove header?

File: runbotics/runbotics-orchestrator-ui/src/pages/blog/post/[slug].tsx [hunk 3]
```diff
@@ -27,27 +22,16 @@ interface Params extends Record<string, string> {
...
-    params,
-    locale,
-}) => {
-    let post: BlogPost | undefined;
+export const getServerSideProps: GetServerSideProps<Props, Params> = async ({ params, locale }) => {
+    const language = locale as Language;
 
-    if (isCacheUpToDate((locale ? locale : 'en') as Language)) {
-        post = getSinglePostCache(locale as Language, params.slug);
-        res.setHeader('X-Cache', 'HIT');
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/pages/blog/post/[slug].tsx:39]: same here

File: runbotics/runbotics-orchestrator-ui/src/pages/index.tsx [hunk 4]
```diff
@@ -16,14 +16,9 @@ interface Props {
 
 const IndexPage: VFC<Props> = ({ blogPosts }) => <MainView blogPosts={blogPosts} />;
 
-export const getServerSideProps: GetServerSideProps<Props> = async ({ res, locale }) => {
-    let cache = getBlogMainCache(locale as Language);
-
-    if (!cache) {
-        cache = await recreateCache(locale as Language);
-    } else {
-        res.setHeader('X-Cache', 'HIT');
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/pages/index.tsx:25]: and here

**Review by preludium — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-orchestrator-ui/src/pages/_app.tsx (hunk 1 above)
Comment (marcin-depcik) [reply runbotics/runbotics-orchestrator-ui/src/pages/_app.tsx:48]: I tested this solution in development and production mode. In development mode, the compilation is executed only when the user enters the page, so this logic is run only when the user visits the page. In production mode, the application runs based on already compiled code, so the logic here is executed after the application is launched without the user having to visit the page

File: runbotics/runbotics-orchestrator-ui/src/pages/blog/index.tsx (hunk 2 above)
Comment (marcin-depcik) [reply runbotics/runbotics-orchestrator-ui/src/pages/blog/index.tsx:63]: image

Above code inside _app.tsx is executed each time user entered the page, so to remove code duplication, I decided to move it.

File: runbotics/runbotics-orchestrator-ui/src/pages/_app.tsx (hunk 1 above)
Comment (marcin-depcik) [reply runbotics/runbotics-orchestrator-ui/src/pages/_app.tsx:48]: I can provide an additional condition to clarify that this logic is intended for `production` purposes only, eg:
`if (process.env.NODE_ENV === 'production' && (!isCached('en') || !isCached('pl')))`

File: runbotics/runbotics-orchestrator-ui/src/pages/blog/index.tsx (hunk 2 above)
Comment (preludium) [reply runbotics/runbotics-orchestrator-ui/src/pages/blog/index.tsx:63]: wrong, we want it to be only on blog related pages

**Review by preludium — CHANGES_REQUESTED** [review]

**Review by preludium — APPROVED** [review]

**Pull request merged**